### PR TITLE
Add 19.1.0-rc3 release

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

In this PR I would like to add llvm 19.1.0rc3 (which should be officially released in a week I think) to the conda-forge. I'm not sure if these are all changes required.

### CODE REVIEW

**Comment by mtsokol** [comment]
Also, `site-packages/mlir/_mlir_libs/include` only has `mlir-c` directory, I think we also need `mlir` there, and `libmlir_c_runner_utils.so` should be available there.

**Comment by mtsokol** [comment]
My guess is in `recipe/build.sh` `cmake` command needs to be updated.

File: recipe/meta.yaml
Comment (hameerabbasi) [inline recipe/meta.yaml:8]:
```suggestion
  - url: https://github.com/llvm/llvm-project/releases/download/llvmorg-{{ version.replace(".rc", "-rc") }}/llvm-project-{{ version.replace(".rc", "-rc") }}.src.tar.xz
```

File: recipe/meta.yaml
Comment (hameerabbasi) [inline recipe/meta.yaml:1]:
```suggestion
{% set version = "19.1.0.rc3" %}
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
You might want to review [the docs for creating a pre-release build for conda-forge](https://conda-forge.org/docs/maintainer/knowledge_base/#creating-a-pre-release-build). It seems like all the missing packages such as [`llvm`](https://anaconda.org/conda-forge/llvm/labels), [`llvmdev`](https://anaconda.org/conda-forge/llvmdev/labels) and [`mlir`](https://anaconda.org/conda-forge/mlir/labels) have the `llvm_rc` label applied, and the version number is `19.1.0.rc3`.

**Comment by hameerabbasi** [comment]
> Also, `site-packages/mlir/_mlir_libs/include` only has `mlir-c` directory, I think we also need `mlir` there, and `libmlir_c_runner_utils.so` should be available there.

This file is probably already in the `mlir` feedstock since it isn't specific to Python.

**Comment by hameerabbasi** [comment]
@mtsokol I opened #35, feel free to pull the changes into your branch if you wish.

**Comment by mtsokol** [comment]
Thank you! It looks like it's building for most environments. Once it installs I will move your commit here.

One question: Should it target `rc` branch or `main` here? 
https://conda-forge.org/docs/maintainer/knowledge_base/#creating-a-pre-release-build says "To create a dev or rc package, a PR can be issued into the dev or rc branch of the feedstock.". I think we don't care about RC for long time, once 19.1.0 is relesed we can also target `main` and override RC configuration here. WDYT?

 
> This file is probably already in the `mlir` feedstock since it isn't specific to Python. It doesn't need to be in `site-packages/mlir/_mlir_libs/include`, but anywhere in `$DYLD_LIBRARY_PATH` should be okay if it's a `.so`.

Exactly! `conda install conda-forge::mlir` installed what I wanted.

**Comment by hameerabbasi** [comment]
> I think we don't care about RC for long time, once 19.1.0 is relesed we can also target `main` and override RC configuration here. WDYT?

We'd have to wait until `llvm`, `llvmdev` and `mlir` also release on `conda-forge` (since they are deps), but we could probably also make that PR given that rc3 is already building, and I don't see a reason why it wouldn't be merged soon.

**Comment by mtsokol** [comment]
Superseded by https://github.com/conda-forge/mlir-python-bindings-feedstock/pull/35

**Pull request closed without merging**
